resolver_test.go

```diff
@@ -91,3 +96,43 @@ func (s) TestResolverCaseSensitivity(t *testing.T) {
 	}
 	cc.Close()
 }
+
+// TestResolverAddressesToEndpoints ensures one Endpoint is created for each
+// entry in resolver.State.Addresses automatically.
+func (s) TestResolverAddressesToEndpoints(t *testing.T) {
+	const scheme = "testresolveraddressestoendpoints"
+	r := manual.NewBuilderWithScheme(scheme)
+
+	stateCh := make(chan balancer.ClientConnState, 1)
+	bf := stub.BalancerFuncs{
+		UpdateClientConnState: func(_ *stub.BalancerData, ccs balancer.ClientConnState) error {
+			stateCh <- ccs
+			return nil
+		},
+	}
+	balancerName := "stub-balancer-" + scheme
+	stub.Register(balancerName, bf)
+
+	a1 := attributes.New("x", "y")
+	a2 := attributes.New("a", "b")
+	r.InitialState(resolver.State{Addresses: []resolver.Address{{Addr: "addr1", BalancerAttributes: a1}, {Addr: "addr2", BalancerAttributes: a2}}})
+
+	cc, err := Dial(r.Scheme()+":///",
+		WithTransportCredentials(insecure.NewCredentials()),
+		WithResolvers(r),
+		WithDefaultServiceConfig(fmt.Sprintf(`{"loadBalancingConfig": [{"%s":{}}]}`, balancerName)))
+	if err != nil {
+		t.Fatalf("Unexpected error dialing: %v", err)
+	}
+	got := <-stateCh
```

Comment:
Nit: In the odd case that this test fails, `select` on the default test timeout as well, so that this does not block for our overall test timeout.

Reply:
Done